feature_engine/imputation/base_imputer.py

```diff
@@ -87,17 +88,46 @@ def _get_feature_names_in(self, X):
 
         return self
 
-    def get_feature_names_out(self) -> List:
+    @Substitution(
+        input_features=_input_features_docstring,
+    )
+    def get_feature_names_out(
+            self, input_features: Optional[Union[List, str]] = None
+    ) -> List:
         """Get output feature names for transformation.
 
+        Parameters
+        ----------
+        {input_features}
+
         Returns
         -------
         feature_names_out: list
             The feature names.
         """
         check_is_fitted(self)
 
-        return self.feature_names_in_
+        if input_features is None:
+            # return all feature names
+            feature_names = [
+                f for f in self.feature_names_in_ if f not in self.features_to_drop_
+            ]
+
+        else:
```

Comment:
we will be repeating this bit of code over and over, I wonder if it would be best to make an abstraction and then just call that function.

Reply:
Make sense. Should it go in variable_manipulation.py?

Reply:
> hola @solegalli,
> 
> What do you mean by `let's not change the docstrings of the classes that create features, and as such, have dedicated get_feature_names_out code`?
> 
> Don't all the transformers create new features?

Nope. Most transformers would transform the original features. Only transformers in the creation, timeseries and datetime modules create new features. And the one hot encoder.

Reply:
Ok, got it. We're all good. The `get_feature_names_out` docstrings for the creation, time-series, and datetime modules and one-hot encoder are the initial, non-generic docstrings.